examples/community/pipeline_flux_rf_inversion.py

```diff
@@ -0,0 +1,969 @@
+        )
+        self.image_latents = image_latents.clone()
+
+        # 2. prepare timesteps
+        sigmas = np.linspace(1.0, 1 / num_inversion_steps, num_inversion_steps)
+        image_seq_len = (int(height) // self.vae_scale_factor // 2) * (int(width) // self.vae_scale_factor // 2)
+        mu = calculate_shift(
+            image_seq_len,
+            self.scheduler.config.base_image_seq_len,
+            self.scheduler.config.max_image_seq_len,
+            self.scheduler.config.base_shift,
+            self.scheduler.config.max_shift,
+        )
+        timesteps, num_inversion_steps = retrieve_timesteps(
+            self.scheduler,
+            num_inversion_steps,
+            device,
+            timesteps,
+            sigmas,
+            mu=mu,
+        )
+        timesteps, sigmas, num_inversion_steps = self.get_timesteps(num_inversion_steps, strength)
+
+        # 3. prepare text embeddings
+        (
+            prompt_embeds,
+            pooled_prompt_embeds,
+            text_ids,
+        ) = self.encode_prompt(
+            prompt=source_prompt,
+            prompt_2=source_prompt,
+            device=device,
+        )
+        # 4. handle guidance
+        if self.transformer.config.guidance_embeds:
+            guidance = torch.full([1], source_guidance_scale, device=device, dtype=torch.float32)
+        else:
+            guidance = None
+
+        # Eq 8 dY_t = [u_t(Y_t) + γ(u_t(Y_t|y_1) - u_t(Y_t))]dt
+        Y_t = image_latents
+        y_1 = torch.randn_like(Y_t)
+        N = len(sigmas)
+
+        # forward ODE loop
+        with self.progress_bar(total=N - 1) as progress_bar:
+            for i in range(N - 1):
+                t_i = torch.tensor(i / (N), dtype=Y_t.dtype, device=device)
+                timestep = torch.tensor(t_i, dtype=Y_t.dtype, device=device).repeat(batch_size)
+
+                # get the unconditional vector field
+                u_t_i = self.transformer(
+                    hidden_states=Y_t,
+                    timestep=timestep,
+                    guidance=guidance,
+                    pooled_projections=pooled_prompt_embeds,
+                    encoder_hidden_states=prompt_embeds,
+                    txt_ids=text_ids,
+                    img_ids=latent_image_ids,
+                    joint_attention_kwargs=self.joint_attention_kwargs,
+                    return_dict=False,
+                )[0]
+
+                # get the conditional vector field
+                u_t_i_cond = (y_1 - Y_t) / (1 - t_i)
+
+                # controlled vector field
+                # Eq 8 dY_t = [u_t(Y_t) + γ(u_t(Y_t|y_1) - u_t(Y_t))]dt
+                u_hat_t_i = u_t_i + gamma * (u_t_i_cond - u_t_i)
+                Y_t = Y_t + u_hat_t_i * (sigmas[i] - sigmas[i + 1])
+                progress_bar.update()
+
+        self.inverted_latents = Y_t
+        self.latent_image_ids = latent_image_ids
+
+        return self.image_latents, Y_t, latent_image_ids
```

Comment:
Curious as to why do we need to assign them to `self`. Maybe we could clarify the order of steps one needs to take to use the inversion pipeline? That would clarify a lot of things.

Reply:
I was mostly following the design/flow we had for ledits++(as it's also an inversion based pipeline) where we keep the inverted latents assigned to self after calling `invert`. But I don't feel strongly about it

Reply:
just popping in to say for purposes in production where we are storing the pipeline in memory in a worker, it's pretty undesirable to have any ephemeral state related to a single request stored on the pipeline instance. I think returning back the latents from the method then passing them back to the __call__ method is much preferable.

Reply:
thank you for considering my comment and making the change! :smiley: